astropy/io/misc/yaml.py

```diff
@@ -110,6 +110,9 @@ def _timedelta_constructor(loader, node):
 
 
 def _ndarray_representer(dumper, obj):
+    if obj.dtype == np.dtype("O"):
```

Comment:
Here, maybe best to use `obj.dtype.hasobject` - this covers structured dtype with object inside it as well.